litgpt/model.py

```diff
@@ -254,10 +289,25 @@ def forward(
     def scaled_dot_product_attention(
         self, q: torch.Tensor, k: torch.Tensor, v: torch.Tensor, mask: Optional[torch.Tensor] = None
     ) -> torch.Tensor:
-        scale = 1.0 / math.sqrt(self.config.head_size)
-        y = torch.nn.functional.scaled_dot_product_attention(
-            q, k, v, attn_mask=mask, dropout_p=0.0, scale=scale, is_causal=mask is None
-        )
+        scale = 1.0 / math.sqrt(self.config.attention_scores_scalar or self.config.head_size)
```

Comment:
The setting we had before

```
scale = 1.0 / math.sqrt(self.config.head_size)
```

isn't that the same as setting `scale=None` in PyTorch's scaled dot product attention? I guess it's good to be explicit, but just curious.

Reply:
Yes of course. No need to change it here. I know you don't like code comments, but perhaps consider adding a comment here so we know in the future

Reply:
later I'll create an issue with a list what also needs to be done.
It will be a long list :)